raku/lib/YAMLScript.rakumod

```diff
@@ -12,21 +12,27 @@ sub load_ys_to_json(uint64, Str --> Str)
 sub graal_create_isolate(uint64 is rw, uint64 is rw, uint64 is rw --> uint64)
   is native('yamlscript', YAMLSCRIPT_VERSION) {*};
 
-has uint64 $!thread;
+sub graal_tear_down_isolate(uint64 --> uint64)
+  is native('yamlscript', YAMLSCRIPT_VERSION) {*};
+
+has uint64 $!isolate-thread;
 
 submethod BUILD {
-  my uint64 ($a, $t, $c);
-  my $rc = graal_create_isolate($a, $t, $c);
+  my uint64 ($n1, $n2);
+  my $rc = graal_create_isolate($n1, $n2, $!isolate-thread);
   die 'Failed to set up evaluation'
     if $rc != 0;
-  $!thread = $c;
+}
+
+submethod DESTROY {
+  my $rc = graal_tear_down_isolate($!isolate-thread);
+  warn 'Failed to destroy isolate thread'
+    unless $rc == 0;
 }
 
 method load(Str $program) {
-  my %res = from-j load_ys_to_json($!thread, $program);
-  if %res<error>:exists {
-    die %res<error><cause>;
-  } else {
-    %res<data>;
-  }
+  my %data-json = from-j load_ys_to_json($!isolate-thread, $program);
+  die %data-json<error><cause>
+    if %data-json<error>:exists;
+  %data-json<data>;
```

Comment:
You should check if `data` key exists and die with an "unknown error" message otherwise.